### PR TITLE
Allows you to create docker image

### PR DESCRIPTION
**Pull request by pollosp**
Hello, 
I have added this in order to use aws-okta from a docker image, I think it is useful specially with windows users.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM golang:1.9-stretch AS builder
+WORKDIR /go/src/github.com/segmentio/aws-okta
+RUN useradd -u 10001 scratchuser
+RUN apt-get update && apt-get -y install libusb-dev  libusb-1.0-0-dev && pkg-config --cflags --libs libusb-1.0
+COPY . .
+RUN make container-build
+
+FROM debian:stretch-slim
+WORKDIR /app/
+RUN apt-get update && apt-get -y install libusb-1.0-0 ca-certificates python-pip && pip install awscli
```
Comment (nickatsegment) [inline Dockerfile]: I wonder if we can statically link `libusb`? Then we'll have a standalone binary and can use a smaller image (both in terms of size and security footprint) like `alpine` or even `scratch`

File: Makefile [hunk 2]
```diff
@@ -13,6 +13,8 @@ test: | govendor
 
 all: dist/aws-okta-$(VERSION)-darwin-amd64 dist/aws-okta-$(VERSION)-linux-amd64
 
+container-build: dist/aws-okta-$(VERSION)-linux-amd64
```
[the remark this answers is not in the document]
Comment (pollosp) [reply Makefile]: Thanks for the comments , this makes senses to me 👍

File: Makefile [hunk 3]
```diff
@@ -27,7 +29,11 @@ dist/aws-okta-$(VERSION)-linux-amd64: | govendor dist/
 	govendor sync
 	GOOS=linux GOARCH=amd64 go build $(LDFLAGS) -o $@
 
+dist/aws-okta-$(VERSION)-win-amd64: | govendor dist/
```
[the remark this answers is not in the document]
Comment (pollosp) [reply Makefile]: Oh yes, you are right :)

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+COPY --from=builder /go/src/github.com/segmentio/aws-okta/dist/aws-okta-*-linux-amd64 /app/aws-okta
+COPY --from=builder /etc/passwd /etc/passwd
```
[the remark this answers is not in the document]
Comment (pollosp) [reply Dockerfile]: Yes I usually use this way with alpine images, maybe I could move to alpine to make this image lighter or just avoid this and use useradd I will check it later.

File: Dockerfile (hunk 1 above)
Comment (pollosp) [reply Dockerfile]: good gotcha let me try with CGO=0 during building time.

[the remark this answers is not in the document]
Comment (pollosp) [reply Dockerfile]: Because I thought that people was going to use this mainly with AWS, I know you can pipe whatever you want but .. it made sense to me.

Comment (pollosp) [reply Dockerfile]: I have trying different combinations I couldn't link libusb statically :S 

```
 CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags '-w -extldflags "-static"' -o mybin *.go
```
And it didn't work any clue here?

[the remark this answers is not in the document]
Comment (nickatsegment) [reply Dockerfile]: I would keep it minimal. If it were a single binary, it'd be fine, but adding all of Python increases the security vulnerability of the image. And `awscli` probably gets released a lot more often than `aws-okta`; I wouldn't want it to lag behind.

How about a second image `aws-okta-toolbox` or something that comes with `awscli` installed?

Comment (nickatsegment) [reply Dockerfile]: Yeah, I don't have any pointers. If you can't get it to work, no worries. Nice to have

[the remark this answers is not in the document]
Comment (pollosp) [reply Dockerfile]: Make sense to me I will do it in that way

**Comment by nickatsegment** [comment]
@pollosp 	Thanks again. We had a Segmenter add https://github.com/segmentio/aws-okta/pull/130, and I hadn't had the chance to circle back to this PR since you added fixes. I'm going to merge #130 for no other reason than we control it. Thanks for your PR though, and sorry I didn't get to it sooner.

**Pull request closed without merging**